packages/babel-helper-explode-assignable-expression/src/index.ts

```diff
@@ -1,12 +1,20 @@
 import type { Scope } from "@babel/traverse";
 import * as t from "@babel/types";
+import type {
+  AssignmentExpression,
+  Identifier,
+  MemberExpression,
+  Literal,
+  Super,
+} from "@babel/types";
 
-function getObjRef(node, nodes, file, scope) {
+function getObjRef(
+  node: Identifier | MemberExpression,
+  nodes: Array<AssignmentExpression>,
+  scope: Scope,
+): Identifier | Super {
   let ref;
-  if (t.isSuper(node)) {
-    // Super cannot be directly assigned so lets return it directly
-    return node;
```

Comment:
This is unreachable per Babel's usage: https://github.com/babel/babel/blob/9891c020b1921136f368426c243156e9fc6fcb29/packages/babel-helper-builder-binary-assignment-operator-visitor/src/index.js#L13

The left of an assignment expression can never be a `Super`.

Even if some one passes a `Super` here, it will throw in
https://github.com/babel/babel/blob/9891c020b1921136f368426c243156e9fc6fcb29/packages/babel-helper-explode-assignable-expression/src/index.js#L47

because `t.cloneNode(node.property)` is `undefined` and the `AssignmentExpression` constructor will reject undefined as input. So we better throw early that `Super` can not be exploded.